src/components/vmd-search.component.ts

```diff
@@ -57,14 +57,14 @@ export class VmdSearchComponent extends LitElement {
     private onCommuneSelected (commune: Commune) {
       this.currentSelection = commune
       this.dispatchEvent(new CustomEvent<SearchRequest.ByCommune>('on-search', {
-        detail: SearchRequest.ByCommune(commune, 'distance', this.currentSearchType || 'standard')
+        detail: SearchRequest.ByCommune(commune, this.currentSearchType || 'standard', this.currentValue ? this.currentValue.date : undefined)
```

Comment:
```suggestion
        detail: SearchRequest.ByCommune(commune, this.currentSearchType || 'standard', this.currentValue?.date)
```

De manière générale, j'évite de passer explicitement `undefined`. Je trouve que ce n'est pas fait pour ça.

Reply:
Je suis d'accord, mais le problème c'est que `this.currentValue` est de type `SearchRequest | void` (et non `SearchRequest | undefined`

=> Cela avait été introduit comme ça par @Floby, et empêche l'utilisation de l'optional chaining 

Je ne sais pas s'il y a un argument en faveur du type `void` par rapport au `undefined` : je n'utilise que très rarement ce type.
@Floby une raison particulière de préférer ce type ?

Reply:
en fait `SearchRequest | void` c'est ce qu'il met quand on déclare le champ avec un point d'intérrogation. Perso void et undefined sont presque synonymes, donc ça me choque pas de changer en `| undefined` j'ai surtout rajouté le type pour qu'on sache que parfois c'est undefined ^^